openpdf/src/main/java/com/lowagie/text/pdf/PdfDocument.java

```diff
@@ -2526,6 +2531,97 @@ protected void add(Image image) throws DocumentException {
+                    float diff = leading / 2;
+                    if (textwrap) {
+                        diff += leading;
+                    }
+                    float lowerleft = footer.getTop() - image.getRelativeTop() - image.getScaledHeight() - diff;
+
+                    float[] mt = image.matrix();
+                    float startPosition = indentLeft() - mt[4];
+                    if ((image.getAlignment() & Image.RIGHT) == Image.RIGHT)
+                        startPosition = indentRight() - image.getScaledWidth() - mt[4];
+                    if ((image.getAlignment() & Image.MIDDLE) == Image.MIDDLE)
+                        startPosition = indentLeft() + ((indentRight() - indentLeft() - image.getScaledWidth()) / 2) - mt[4];
+                    if (image.hasAbsoluteX()) startPosition = image.getAbsoluteX();
+
+                    if (!textwrap) {
+                        if ((image.getAlignment() & Image.RIGHT) == Image.RIGHT)
+                            startPosition -= image.getIndentationRight();
+                        else if ((image.getAlignment() & Image.MIDDLE) == Image.MIDDLE)
+                            startPosition += image.getIndentationLeft() - image.getIndentationRight();
+                        else startPosition += image.getIndentationLeft();
+                    }
+                    graphics.addImage(image, mt[0], mt[1], mt[2], mt[3], startPosition, lowerleft - mt[5]);
+                    break;
+                }
+            }
+        }
+        footer.setPadding(0);
+    }
+
+    /**
+     * Occupies space for <CODE>Image</CODE> that will be added later instead of now
+     *
+     * @param image the new <CODE>Image</CODE>
+     */
+
+    protected void addDelay(Image image) {
+        if (image.hasAbsoluteY()) {
+            System.out.println("Warning: absoluteY of image is invalid in footer");
+        }
+
+        image.setRelativeTop(currentHeight); // set the offset relative to the top
+        image.setAlignment(image.getAlignment() | footer.alignment());
+        footer.addSpecialContent(image);
+
+        // add indentation for text
+        boolean textwrap = (image.getAlignment() & Image.TEXTWRAP) == Image.TEXTWRAP
+                && !((image.getAlignment() & Image.MIDDLE) == Image.MIDDLE);
+        boolean underlying = (image.getAlignment() & Image.UNDERLYING) == Image.UNDERLYING;
+        float diff = leading / 2;
+        if (textwrap) {
+            if (imageEnd < 0 || imageEnd < currentHeight + image.getScaledHeight() + diff) {
+                imageEnd = currentHeight + image.getScaledHeight() + diff;
+            }
+            if ((image.getAlignment() & Image.RIGHT) == Image.RIGHT) {
+                // indentation suggested by Pelikan Stephan
+                indentation.imageIndentRight += image.getScaledWidth() + image.getIndentationLeft();
+                indentation.imageIndentRight += image.getScaledWidth() + image.getIndentationLeft();
```

Comment:
@ObsisMc is adding the indentation twice intentional? If not, I can remove it as part of my table sorcery.

Reply:
sorry, I think it is a mistake